craft_application/models/project.py

```diff
@@ -52,6 +52,25 @@ class BuildInfo:
     """The base to build on."""
 
 
+class BuildPlannerConfig(CraftBaseConfig):
+    """Config for BuildProjects."""
+
+    extra = pydantic.Extra.ignore
+    """The BuildPlanner model uses attributes from the project yaml."""
+
+
+class BuildPlanner(CraftBaseModel):
```

Comment:
I like moving this off the project, but before the 2.0 release I'd like to discuss where to actually put the `get_build_plan` method. Out of scope here, but it's something that doesn't quite feel like it has an obvious home.